### PR TITLE
(MODULES-9444) Migrate puppet_agent module to Beaker 4

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -40,15 +41,13 @@ def install_modules_on(host)
   options['is_puppetserver'] = true
   master['puppetservice'] = 'puppetserver'
   master['puppetserver-confdir'] = '/etc/puppetlabs/puppetserver/conf.d'
-  master['type'] = 'aio'
-  install_puppet_agent_on master, {}
-  install_package master, 'puppetserver'
-  master['use-service'] = true
+  master['type'] = 'foss'
```
Comment (gimmyxd) [inline spec/spec_helper_acceptance.rb]: Is breaking if we use `aio`?
https://github.com/puppetlabs/beaker/blob/master/docs/concepts/types_puppet_4_and_the_all_in_one_agent.md

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -40,15 +41,13 @@ def install_modules_on(host)
   options['is_puppetserver'] = true
   master['puppetservice'] = 'puppetserver'
   master['puppetserver-confdir'] = '/etc/puppetlabs/puppetserver/conf.d'
-  master['type'] = 'aio'
-  install_puppet_agent_on master, {}
-  install_package master, 'puppetserver'
```
Comment (gimmyxd) [inline spec/spec_helper_acceptance.rb]: Is server installed by beaker-puppet now?

Comment (oanatmaria) [reply spec/spec_helper_acceptance.rb]: puppet-server is not installed on master neither on host, as the server is not necessary in the context of these tests

**Comment by oanatmaria** [comment]
Tests are passing: https://jenkins-master-prod-1.delivery.puppetlabs.net/job/forge-module_puppetlabs-puppet-agent-module_intn-sys_smoke-adhoc/44/

File: spec/spec_helper_acceptance.rb [hunk 3]
```diff
@@ -223,7 +216,6 @@ def teardown_puppet_on(host)
   else
     pp = <<-EOS
 #{clean_repo}
-file { ['/etc/puppet', '/etc/puppetlabs', '/etc/mcollective']: ensure => absent, force => true, backup => false }
```
Comment (gimmyxd) [inline spec/spec_helper_acceptance.rb]: why are those files still kept on teardown?

**Pull request merged**